cfme/services/requests.py

```diff
@@ -1,288 +1,496 @@
-
-    Usage:
-
-        # Filter on the "Description" column
-        description = 'Provision from [%s] to [%s]' % (template_name, vm_name)
-        cells = {'Description': description}
-
-        # Filter on the "Request ID" column
-        # Text must match exactly, you can use "{:,}".format(request_id) to add commas if needed.
-        request_id = '{:,}'.format(1000000000001)  # Becomes '1,000,000,000,001', as in the table
-        cells = {'Request ID': request_id}
-
-        # However you construct the cells dict, pass it to wait_for_request
-        # Provisioning requests often take more than 5 minutes but less than 10.
-        wait_for(wait_for_request, [cells], num_sec=600)
-
-    Raises:
-        RequestException: if multiple matching requests were found
-
-    Returns:
-         The matching :py:class:`cfme.web_ui.Table.Row` if found, ``False`` otherwise.
-    """
-    row = find_request(cells, partial_check)
-    if row.request_state.text in REQUEST_FINISHED_STATES:
-        return row
-    else:
-        return False
-
-
-def debug_requests():
-    logger.debug('Outputting current requests')
-    from cfme.web_ui import paginator
-    for page in paginator.pages():
-        for row in fields.request_list.rows():
-            logger.debug(' %s', row)
-
-
-def find_request(cells, partial_check=False):
-    """Finds the request and returns the row element
-
-    Args:
-        cells: Search data for the requests table.
-        partial_check: If to use the ``in`` operator rather than ``==`` in find_rows_by_cells().
-    Returns: row
-    """
-    navigate_to(Request, 'All')
-    from cfme.web_ui import paginator
-    for page in paginator.pages():
-        results = fields.request_list.find_rows_by_cells(cells, partial_check)
-        if len(results) == 0:
-            # row not on this page, assume it has yet to appear
-            # it might be nice to add an option to fail at this point
-            continue
-        elif len(results) > 1:
-            raise RequestException(
-                'Multiple requests with matching content found - be more specific!'
-            )
-        else:
-            # found the row!
-            row = results[0]
-            logger.debug(' Request Message: %s', row.last_message.text)
-            return row
-    else:
-        # Request not found at all, can't continue
-        return False
-
-
-def go_to_request(cells, partial_check=False):
-    """Finds the request and opens the page
-
-    See :py:func:`wait_for_request` for futher details.
-
-    Args:
-        cells: Search data for the requests table.
-    Returns: Success of the action.
-    """
-    row = find_request(cells, partial_check)
-    if row:
-        sel.click(row)
-        return True
-    else:
-        return False
-
-
-@contextmanager
-def copy_request(cells):
-    """Context manager that opens the request for editing and saves or cancels depending on success.
+    def am_i_here(self, *args, **kwargs):
```

Comment:
**REQUIRED**
Here and further please implement `is_displayed` in a respectful view and remove `am_i_here`